### PR TITLE
Support for static IP configuration in azurerm_private_endpoint #13061

### PR DESCRIPTION
**Pull request by Noel-Jones**
Added IP Configuration block. I only allow one to be configured despite the implication that a private endpoint can be created for multiple subresources and an ip could be configured for each. In practice it is not possible to specify a list of subresources anywhere in the Azure Console that I have seen. Tested within my sandbox environment; creation with static IP and recreation from dynamic IP (no IP Configuration) to static IP.

### CODE REVIEW

File: website/docs/r/private_endpoint.html.markdown [hunk 1]
```diff
@@ -239,6 +251,16 @@ A `private_service_connection` block exports:
 
 ---
 
+An `ip_configuration` block exports:
```
Comment (catriona-m) [inline website/docs/r/private_endpoint.html.markdown:254]: it looks like this has been added to the docs twice?

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for submitting this @Noel-Jones, looking pretty good so far, however it would be good if we could add tests to confirm the new block works as expected.

Also have one minor query on the docs. Thanks!

File: website/docs/r/private_endpoint.html.markdown (hunk 1 above)
Comment (lmarkusl) [reply website/docs/r/private_endpoint.html.markdown:254]: Hi @catriona-m isn't it one for set and the other ones which values you get back?

**Review by katbyte — APPROVED** [review]
LGTM 🧄

**Pull request merged**